Improve: allow profiles to be individually resized when multi-playing

SlySven opened the pull request:
This is for when multi-view mode is activated.

It was suggested in our [Discord **#help** channel recently](https://www.discord.com/channels/283581582550237184/283582068334526464/918258911289696257) but in fact is a *Launchpad* issue first raised in 2013! As such this PR should close #671...!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

#### Release post highlight
Improvement: when multi-playing and *multiview* mode is active the horizontal space allocated to each profile can now be adjusted by dragging the separator between each profile's main window.

SlySven committed:
Improve: allow profiles to be individually resized when multi-playing

This is for when multi-view mode is activated.

It was suggested in our Discord **#help** channel recently:
https://www.discord.com/channels/283581582550237184/283582068334526464/918258911289696257
but in fact is a *Launchpad* issue first raised in 2013! As such this PR
should close #671...!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented on src/mudlet.h:
```diff
@@ -624,6 +625,7 @@ private slots:
 
     QWidget* mpWidget_profileContainer;
     QHBoxLayout* mpHBoxLayout_profileContainer;
+    QSplitter* mpSpltter_profileContainer;
```
Spltter?

SlySven replied:
We're both wrong - should be "sp**LI**tter"...! 😆

demonnic commented:
worked well for me, tested with 3 profiles open at once and noticed no interactions with the splitters which seemed unexpected.

SlySven commented:
> ... and noticed no interactions with the splitters which seemed unexpected.

Can you explain what you mean by that - and if it is good or bad? :confused:

demonnic commented:
It all worked as expected.

vadi2 commented on src/mudlet.cpp:
```diff
@@ -4597,30 +4604,27 @@ void mudlet::slot_tabMoved(const int oldPos, const int newPos)
-            layoutItemMap.insert(name, pLayoutItem);
+            widgetMap.insert(name, pWidget);
+        } else {
+            qWarning().nospace().noquote() << "mudlet::slot_tabMoved(" << oldPos<< ", " << newPos << ") WARNING - nullptr for pointer to TMainConsole at 'profileIndex': " << profileIndex << ".";
```
If one sees this warning, it'll make no sense at all. Should we still display it?

SlySven replied:
Well, if it happens it means the whole profile that was at "oldPos" in the tab bar hole profile has suddenly disappeared when it was being moved to "newPos" - so some sort of diagnostic output is going to help with a post-mortem...

SlySven replied:
Also, as that line uses method's arguments I don't need the pair of `Q_UNUSED(`...`)` at the start of the slot...! 😜

SlySven committed:
Fix: correct a member name from mpSpltter... to mpSplitter...

Also remove a pair of `Q_UNUSED()` macros as the arguments they work on
*are* used in a debugging message.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

atari2600tim commented:
Appears to work, even the NAWS is working for both.
Noticed some odd stuff but I think it is all pre-existing.
